BrainCloudClient/Assets/BrainCloud/Client/BrainCloud/Internal/BrainCloudComms.cs

```diff
@@ -2265,6 +2254,27 @@ private void DisposeUploadHandler()
             }
 #endif
         }
+
+        public void AddCallbackToAuthenticateRequest(ServerCallback in_callback)
+        {
+            //Authenticate requests will always be put to the top of the request
+            _serviceCallsInProgress[0].GetCallback().AddCallbacks(in_callback);
```

Comment:
Iterate through the list of calls similar to the IsAuthenticateRequestInProgress, JUUST in case something else gets into the queue. Yes it should be at the front, but I wouldn't always rely on it. I'd iterate through those in progress calls which are authenticate, and add to the callback like you are doing.

Reply:
Funny, I had that previously and thought this was the better approach, but you're right with we shouldn't always rely on that. I'll change it to how you suggested.